### ISSUE TITLE
Create system table with server parameters

### ISSUE THREAD

**Issue by qoega** [ClickHouse/ClickHouse#13924]
**Use case**
We want to read sever settings like `dns_cache_update_period` from SQL.

**Describe the solution you'd like**
Create a wrapper that reads setting value and registers it in system table when used.
For example instead of 
``` c++
config().getInt("dns_cache_update_period", 15)
```

We can have ConfigReader class with interface like 
``` c++
getInt(std::string path, int32 default, std::string description, bool hide_value, std::string scope)
```
When we call it we can add data to system table on startup.
Description can be used as one in system.settings. 

`scope` can be used to debug which parts of server use this setting if we don't want to ban usage of same setting from different server parts.

We may need to hide credentials.

There could be settings available in config but not used in server - we will not see them. It should be OK.

**Comment by filimonov** [comment]
https://github.com/ClickHouse/ClickHouse/issues/7663 ?

**Comment by alexey-milovidov** [comment]
That's a good idea :+1: 
To "publish" subset of config parameters that were in effect.

@filimonov 
It's much better than simply export all the configuration.

**Comment by filimonov** [comment]
Agree, added a link to another issue just to keep the issues related to the same subject together. 

May be there are some corner cases when seeing the raw canfig may be still interesting, but subset of effective values used in code is much more important for users/admins.

Btw maybe we can also store last access time to config element.

**Comment by alexey-milovidov** [comment]
Implemented in #35878 (but in very insecure fashion).
